File: gateway/src/main/java/io/kuz/ecom/gateway/mapper/PaginationMetaMapper.java
package io.kuz.ecom.gateway.mapper;

import io.kuz.ecom.gateway.dto.PaginationMetaDTO;
import io.kuz.ecom.proto.common.PaginationMeta;

public class PaginationMetaMapper {

    public static PaginationMetaDTO toDTO(PaginationMeta proto) {
        return new PaginationMetaDTO(
            (long) proto.getTotal(),
            (long) proto.getLimit(),
            (long) proto.getOffset()
        );
    }
}

File: src/infra/grpc-mapper/src/main/kotlin/io/kuz/grpc/mapper/common/PaginationMetaMapper.kt
package io.kuz.grpc.mapper.common

import io.kuz.ecom.common.models.common.PaginationMetaModel
import io.kuz.ecom.proto.common.PaginationMeta

object PaginationMetaMapper {

    fun toGRPC(model: PaginationMetaModel): PaginationMeta {
        return PaginationMeta
            .newBuilder()
            .setPage(model.page.toInt())
            .setPageSize(model.pageSize.toInt())
            .setPageCount(model.pageCount.toInt())
            .build()
    }

    fun toModel(grpc: PaginationMeta): PaginationMetaModel {
        return  PaginationMetaModel(
            page = grpc.page.toLong(),
            pageSize = grpc.pageSize.toLong(),
            pageCount = grpc.pageCount.toLong(),
        )
    }
}
